common/src/java/org/apache/hadoop/hive/common/auth/HiveAuthUtils.java

```diff
@@ -50,45 +49,108 @@
 public class HiveAuthUtils {
   private static final Logger LOG = LoggerFactory.getLogger(HiveAuthUtils.class);
 
+  /**
+   * Configure the provided T transport's max message size.
+   * @param transport Transport to configure maxMessage for
+   * @param maxMessageSize Maximum allowed message size in bytes, les than or equal to 0 means use the Thrift library
+   *                       default.
+   * @return The passed in T transport configured with desired max message size. The same object passed in is returned.
+   */
+  public static <T extends TTransport> T configureThriftMaxMessageSize(T transport, int maxMessageSize) {
+    if (maxMessageSize > 0) {
+      if (transport.getConfiguration() == null) {
+        LOG.warn("TTransport {} is returning a null Configuration, Thrift max message size is not getting configured",
```

Comment:
Decided to be paranoid here. My initial test run caught TFilterTransport returning null for getConfiguration() (so I plumbed it through in this patch). I figured it is better to be safe than sorry, since an exception in this path likely makes the tables not accessible.

Reply:
makes sense